core/storage/auth/gae_models_test.py

```diff
@@ -72,6 +78,80 @@ def test_get_deletion_policy_is_delete_at_end(self) -> None:
             auth_models.UserAuthDetailsModel.get_deletion_policy(),
             base_models.DELETION_POLICY.DELETE_AT_END)
 
+    def test_get_field_names_for_takeout(self) -> None:
+        expected_dict = {
+            'parent_user_id': 'parent_username'
+        }
+        self.assertEqual(
+            auth_models.UserAuthDetailsModel
+                .get_field_names_for_takeout(),
+            expected_dict)
+
+    def test_get_model_association_to_user(self) -> None:
+        self.assertEqual(
+            auth_models.UserAuthDetailsModel.
+                get_model_association_to_user(),
+            base_models.MODEL_ASSOCIATION_TO_USER.ONE_INSTANCE_PER_USER)
+
+    def test_get_export_policy(self) -> None:
+        expected_export_policy_dict = {
+            'created_on': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'deleted': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'last_updated': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'gae_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'firebase_auth_id': base_models.EXPORT_POLICY.NOT_APPLICABLE,
+            'parent_user_id': base_models.EXPORT_POLICY.EXPORTED,
+        }
+        self.assertEqual(
+            auth_models.UserAuthDetailsModel
+                .get_export_policy(),
+            expected_export_policy_dict)
+
+    def test_export_data_trivial(self) -> None:
+        """Trivial test of export_data functionality."""
+
+        exported_model = (
+            auth_models.UserAuthDetailsModel.export_data(
+                self.NONEXISTENT_USER_ID))
+
+        self.assertEqual(exported_model, {})
+
+    def test_export_data_nontrivial(self) -> None:
+        user_auth_model = (
+            auth_models.UserAuthDetailsModel
+                .get_by_id(self.PROFILE_2_ID))
+        self.assertIsNotNone(user_auth_model)
+
+        # The parent_user_id should exist to fetch the
+        # parent_model further on in the test.
+        self.assertIsNotNone(user_auth_model.parent_user_id)
+
+        # Create the model instance to be fetched using
+        # user_auth_model.parent_user_id. The fetched
+        # parent_model will provide the username to be
+        # returned by .export_data().
+        user_models.UserSettingsModel(
+            id=self.USER_ID,
+            email='user@example.com',
+            roles=[feconf.ROLE_ID_CURRICULUM_ADMIN],
+            banned=False,
+            username='user'
+        ).put()
+
+        parent_model = (
+            user_models.UserSettingsModel.
+                get(user_auth_model.parent_user_id)
+        )
+
+        # Ruling out the possibility of None for mypy type checking.
+        assert parent_model is not None
+        exported_model = (
```

Comment:
This is not a "model" -- it's just a dict.

Reply:
Done!